hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Dispatcher.java

```diff
@@ -1377,6 +1403,22 @@ public void shutdownNow() {
     }
   }
 
+  @Override
+  public void close() throws IOException {
+    storageGroupMap.clear();
```

Comment:
one thing here -- i think when i wrote our original close method, i tried to find any state in the Dispatcher that might make sense to clear/reset/stop. I assume the below is still all inclusive, but might be worth a quick check incase anything was added since cdh5, if you haven't already.

Reply:
I'll take a look. I basically took the contents of `reset()` and copied them over. I was tempted to just use `reset()` here, but I wasn't sure if we wanted to avoid setting the conf to the same thing again.

Reply:
Confirmed, this has the contents of `reset()` and `shutdownNow()`. This should be all inclusive.